File: src/dk/eightyplus/annoto/fragment/SliderFragment.java
package dk.eightyplus.annoto.fragment;

import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SeekBar;
import dk.eightyplus.annoto.Callback;
import dk.eightyplus.annoto.Keys;
import dk.eightyplus.annoto.R;

/**
 * Fragment for width slider
 */
public class SliderFragment extends DialogFragment {

  private Callback callback;
  private int width;

  @SuppressWarnings("unused")
  public SliderFragment() { }

  public SliderFragment(final Callback callback, int width) {
    this.callback = callback;
    this.width = width;
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    if (savedInstanceState != null) {
      width = savedInstanceState.getInt(Keys.WIDTH);
    }

    callback = (Callback) getActivity();
  }

  @Override
  public void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    outState.putInt(Keys.WIDTH, width);
  }

  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.slider, container, false);

    SeekBar slider = (SeekBar) view.findViewById(R.id.slider);
    slider.setProgress(width);
    slider.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
      @Override
      public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) { }

      @Override
      public void onStartTrackingTouch(SeekBar seekBar) { }

      @Override
      public void onStopTrackingTouch(SeekBar seekBar) {
        callback.setStrokeWidth(seekBar.getProgress());
      }
    });

    return view;
  }
}

File: app/src/main/java/dk/eightyplus/annoto/fragment/SliderFragment.kt
package dk.eightyplus.annoto.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SeekBar
import dk.eightyplus.annoto.Callback
import dk.eightyplus.annoto.Keys
import dk.eightyplus.annoto.R
import java.lang.ref.SoftReference

/**
 * Fragment for width slider
 */
class SliderFragment : DialogFragment {

    private lateinit var callbackSoftReference: SoftReference<Callback>
    private var width: Int = 0

    constructor() {}

    @SuppressLint("ValidFragment")
    constructor(callback: Callback, width: Int) {
        this.callbackSoftReference = SoftReference(callback)
        this.width = width
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (savedInstanceState != null) {
            width = savedInstanceState.getInt(Keys.WIDTH)
        }

        this.callbackSoftReference = SoftReference(activity as Callback)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt(Keys.WIDTH, width)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.slider, container, false)

        val slider = view.findViewById<View>(R.id.slider) as SeekBar
        slider.progress = width
        slider.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {}

            override fun onStartTrackingTouch(seekBar: SeekBar) {}

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                callbackSoftReference.get()?.setStrokeWidth(seekBar.progress)
            }
        })

        return view
    }
}
